components/hybrid_engine/src/snapshot.rs

```diff
@@ -68,18 +66,36 @@ where
     EK: KvEngine,
     EC: RangeCacheEngine,
 {
-    type Iterator = HybridEngineIterator<EK, EC>;
+    // TODO (afeinberg): use `Either` instead of `Box<dyn Iterator>`
```

Comment:
Why not implement Either directly so that we dont need Box.

Reply:
done.